### DISCUSSION TITLE
Static IP After Install

### DISCUSSION THREAD

**Discussion by MWalton01** [tteck/Proxmox#1352] [Help Needed] [answered]
I have used the script to create a HA VM and it seems to have worked, however, I can't seem to set a static IP.

I have tried following a number of guides, including [https://github.com/home-assistant/operating-system/blob/7e702a2ab2ee7e76ed7f3a1be6f98a3c832b557e/Documentation/network.md#using-nmcli-to-set-a-static-ipv4-address](url)

However, after I run `nmcli con edit "HassOS default"` (replacing the name with the one relevant to my system of course), it ends with `Segmentation fault (core dumped)` and wont allow me to continue and actually set the IP.

**Comment by tteck** [comment]
<a href="https://github.com/tteck/Proxmox/blob/main/USER_SUBMITTED_GUIDES.md"><img src="https://img.shields.io/badge/%F0%9F%A7%AD-Guides-yellow" alt="Guides"/></a> 

Home Assistant: Proxmox Quick Start Guide has a "Setting Static IP Address" section.

**Reply by MWalton01** [reply]
This does not work as the section requires you to first have access to the panel which I do not. Before its given access to the internet, I need to assign a static IP and gateway as there is no DHCP on Proxmox

**Reply by tteck** [reply]
You can't reach Settings -> System -> Network?

![Screenshot 2023-04-20 12 56 42 PM](https://user-images.githubusercontent.com/21342670/233436746-7595848b-ad08-4bc6-a52f-15b00c10aed7.png)

**Reply by MWalton01** [reply]
No as at current, since there's no DHCP for Proxmox, HA doesn't have any IP, Gateway or DNS set, and when I try to follow the guide I mentioned ([link](https://github.com/home-assistant/operating-system/blob/7e702a2ab2ee7e76ed7f3a1be6f98a3c832b557e/Documentation/network.md#using-nmcli-to-set-a-static-ipv4-address)), it doesn't work as I get an error saying `segmentation fault (core dumped)`

**Reply by tteck** [reply]
Is it not possible for your router to function as the DHCP server?

**Reply by MWalton01** [reply]
> Is it not possible for your router to function as the DHCP server?

Well it does for every other device in my house, it just cant seem to detect or assign IPs to VMs on Proxmox; so instead they are on a subnet going through the IP of proxmox itself and this works fine but requires specifying the IP and Gateway before it can connect to wifi

**Reply by MWalton01** [reply, marked as answer]
disregard, found away to get DHCP to work finally so this point is mute
